### PR TITLE
Add JSHint to default rake task

### PR DESCRIPTION
**Pull request by neoeno**
I also factored out all the puts because, well, I don't know.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -10,15 +10,25 @@ include FileUtils
...
+end
 
+described_task :karma, description: 'Javascript Karma specs' do
   fail 'ERROR: karma is not installed' unless File.exist? KARMA_COMMAND
   sh "#{KARMA_COMMAND} start spec/javascripts/karma.conf.js --single-run=true"
+end
 
-  puts
+described_task :jshint, description: 'JSHint' do
+  sh 'node_modules/.bin/jshint app/assets/javascripts spec/javascripts' do
```
Comment (munckymagik) [inline Rakefile]: Can we factor out the binary existence check here too?

Comment (neoeno) [inline Rakefile]: not sure about factoring it out — but you're right it should definitely be there

**Comment by benbarnett** [comment]
Always wonder where these things should live with Rails projects. Its common to keep scripts like this in the npm package.json (like the [one in Dough](https://github.com/moneyadviceservice/dough/blob/master/package.json#L7)), but then you'd have duplication. Or does one take the canonical list and execute commands using the other? (e.g. rake runs `npm run-script` type commands). Who knows.

Either way, this makes sense for RAD as the rest of these things are in the Rakefile.

:+1:

**Comment by JiggyPete** [comment]
:+1:

File: Rakefile (hunk 1 above)
Comment (munckymagik) [inline Rakefile]: Well, factor it in I mean :)

**Comment by neoeno** [comment]
I like `test.sh` as it's flexible and agnostic, but doesn't have as much meaning outside of Go-bound projects...

That last commit work for you, @munckymagik?

**Comment by munckymagik** [comment]
:+1:

**Comment by neoeno** [comment]
I shouldn't have pressed that button

**Comment by benbarnett** [comment]
When autosquash isn't auto enough...

**Pull request merged**